addons/survey/static/src/views/widgets/integer_update_flag_field/integer_update_flag_fields.js

```diff
@@ -45,18 +47,27 @@ export class IntegerUpdateFlagField extends IntegerField {
             },
             () => [inputRef.el]
         );
+        if (this.referenceValue === undefined) {
+            onWillStart(async () => {
+                const result = await this.orm.searchRead(
+                    "survey.survey",
+                    [["id", "=", this.props.record.data.survey_id[0]]],
+                    ["session_speed_rating_time_limit"]
+                );
+                this.referenceValue = result[0]["session_speed_rating_time_limit"];
```

Comment:
Meh ! If we do that, the widget will no longer be generic and will be specific to the `survey.question` model.

However, I'm not sure we have any better solution for the stable versions. As you mentioned in the task description, we could indeed add related fields to load `session_speed_rating_time_limit` and `session_speed_rating` from the `survey.question` model but the XML view won't be updated when applying the fix on the existing odoo instances.

It should hence be okay to do that given that those widgets are only used in the `survey_question_form` form view.

(same remark for the other widget)

Reply:
:boar: Better to keep everything simple and explicit.